### ISSUE TITLE
Create popular hosting examples

### ISSUE THREAD

**Issue by codeluggage** [codeluggage/imba-static-site#1]
- [ ] Github Pages
- [ ] Netlify
- [ ] Vercel
- [ ] Others?

**Comment by familyfriendlymikey** [comment]
# Deploying To Github Pages

## TLDR
Create a new github repository and replace `YOUR_NEW_GITHUB_REPO_URL` with your new repo's url.
```
npx imba create project_name
cd project_name
git add --all
git commit -m "initial commit"
git remote add origin YOUR_NEW_GITHUB_REPO_URL
git push -u origin main

npx imba build --baseurl . server.imba
npx touch dist/public/.nojekyll
npx gh-pages --no-history --dotfiles --dist dist/public
```
To find the URL your project has been deployed to, navigate to the `Pages` tab of your repo's settings.

## Explanation
For static hosting, we build using `.` as the baseurl.
```
npx imba build --baseurl . server.imba
```
Yes, we still build using the `server.imba` file even though we won't be using any of the server-side files.

Since Github Pages uses Jekyll by default, paths starting with underscores (like `__assets__`) will fail to load, so we have to that specify we don't want to use Jekyll by doing:
```
npx touch dist/public/.nojekyll
```
Once all of that is settled, actually deploying to github pages is really easy with `npx gh-pages`, which will create a new git branch named `gh-pages` and serve our files from there by default.

- For our purposes, it's important to specify `--dotfiles` because of the necessary `.nojekyll` file.
- We also want to use `dist/public` as the base directory.
- Since we don't need the `gh-pages` branch for actual version control, I prefer to use the `--no-history` flag as well.

```
npx gh-pages --no-history --dotfiles --dist dist/public
```
